packages/codegen-sdk/src/parse-operation.ts

```diff
@@ -60,10 +63,24 @@ export function parseOperations(
   const operations = getOperations(documents);
 
   return operations.reduce<SdkDefinitions>((acc, node) => {
-    const path = (node.name?.value ?? "").split("_");
+    const nodeName = node.name?.value ?? "";
+    let path;
+
+    if (
+      node.operation === "mutation" &&
+      hasBeenTransformed(
+        nodeName,
+        context.mutations.map(mutationNode => mutationNode.name.value)
+      )
+    ) {
+      path = restoreSuffixedMutationName(nodeName).split("_");
```

Comment:
Why does it need to be restored again?

Reply:
I think "restore" is a poor naming choice, I will change it. 

When we generate the GraphQL Document out of the GraphQL schema, we name each mutation with a prefixed name, in order to generate a valid Document. But these prefixed mutations still need to point to the proper mutations on the GraphQL server, as the server doesn't know about the prefixed mutations.

For instance, here is the `apiKeyCreate` mutation as it is generated in the Document (minus the comments for readability):

```graphql
mutation createApiKey(

  $input: ApiKeyCreateInput!
) {

  apiKeyCreate(input: $input) {
    ...ApiKeyPayload
  }

}
```

From an SDK perspective, we will be building methods for the `createApiKey` mutation that will end up performing the `apiKeyCreate` mutation on the server. 

So we are not "restoring" the suffixed mutation name in the sense that we would get rid of the prefixed one in favor of the suffixed one. Instead, we're just reverting the previous operation of prefixing, temporarily, in order to find matching mutations and selection fields in the nodes selection sets.

Reply:
Gotcha. 👍